### PR TITLE
Remove VariableVisitConsumerAction.CONSUME

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

See discussion in https://github.com/PyCQA/pylint/pull/5727#discussion_r795078071. Wait for #5727, then rebase and open for review.

### CODE REVIEW

File: pylint/checkers/variables.py [hunk 1]
```diff
@@ -1522,21 +1514,22 @@ def _check_consumer(
...
-                # If there are nodes added to consumed_uncertain by
-                # get_next_to_consume() because they might not have executed,
-                # return a CONSUME action so that _undefined_and_used_before_checker()
-                # will mark them as used
-                return (VariableVisitConsumerAction.CONSUME, found_nodes)
-            return (VariableVisitConsumerAction.RETURN, None)
+            # If there are nodes added to consumed_uncertain by
+            # get_next_to_consume() because they might not have executed,
+            # return a CONSUME action so that _undefined_and_used_before_checker()
+            # will mark them as used
```
Comment (jacobtylerwalls) [inline pylint/checkers/variables.py:1520]: Note to self: when rebasing, fix this comment

File: pylint/checkers/variables.py [hunk 2]
```diff
@@ -1483,7 +1465,6 @@ def _check_consumer(
         consumer_level: int,
         base_scope_type: Any,
     ) -> VariableVisitConsumerActionAndOptionalNodesType:
```
Comment (DanielNoord) [inline pylint/checkers/variables.py:1467]: Why did you remove the docstring here?

File: pylint/checkers/variables.py [hunk 3]
```diff
@@ -175,23 +169,14 @@ class VariableVisitConsumerAction(Enum):
...
-        Union[
-            Literal[VariableVisitConsumerAction.CONTINUE],
-            Literal[VariableVisitConsumerAction.RETURN],
-        ],
-        None,
-    ],
-    Tuple[Literal[VariableVisitConsumerAction.CONSUME], List[nodes.NodeNG]],
+
+
+VariableVisitConsumerActionAndOptionalNodesType = Tuple[
```
Comment (DanielNoord) [inline pylint/checkers/variables.py:178]: I wonder if this is necessary. With the simplification wouldn't it be better to use this directly instead of an alias?

File: pylint/checkers/variables.py (hunk 2 above)
Comment (jacobtylerwalls) [reply pylint/checkers/variables.py]: Oops, that was an oversight. Thanks for catching.

File: pylint/checkers/variables.py (hunk 3 above)
Comment (jacobtylerwalls) [reply pylint/checkers/variables.py]: Yes, I was starting to think that myself.

File: pylint/checkers/variables.py [hunk 4]
```diff
@@ -1526,21 +1503,20 @@ def _check_consumer(
...
-                # get_next_to_consume() because they might not have executed,
-                # return a CONSUME action so that _undefined_and_used_before_checker()
-                # will mark them as used
-                return (VariableVisitConsumerAction.CONSUME, found_nodes)
-            return (VariableVisitConsumerAction.RETURN, None)
+            # Mark for consumption any nodes added to consumed_uncertain by
+            # get_next_to_consume() because they might not have executed.
+            return (
+                VariableVisitConsumerAction.RETURN,
+                current_consumer.consumed_uncertain[node.name],
```
Comment (DanielNoord) [inline pylint/checkers/variables.py:1510]: Is this the same as `current_consumer.get_next_to_consume(node)`?

Comment (jacobtylerwalls) [reply pylint/checkers/variables.py:1510]: I don't think so: that function has already run and actually was responsible for _populating_ the dict entry being accessed here. Now having received a RETURN action, it's time to finish consuming all the names, including what maybe-executed.

I'm realizing `consumed_uncertain` is less descriptive as a name now that we're getting rid of CONSUME as an action. It now refers more to "execution uncertain" or something.

**Review by DanielNoord — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
Nice refactoring !
